sound/soc/sof/intel/hda-loader.c

```
@@ -38,7 +38,7 @@ static int cl_stream_prepare(struct snd_sof_dev *sdev, unsigned int format,
 
 	if (!dsp_stream) {
 		dev_err(sdev->dev, "error: no stream available\n");
-		return -ENODEV;
+		return NULL;
```

I think I'd use `ERR_PTR()` & `PTR_ERR()` with this change to avoid losing exact error codes

---

done!

---

@ranj063 no, sorry, that isn't what I meant. Here you're actually ignoring the error, by returning NULL you're actually returning 0, so `IS_ERR()` will treat this as a success. You need to `return ERR_PTR(-ENODEV);` here.

---

@ranj063 `return ERR_PTR(-ENODEV);`

---

hmm missed that one. fixed now.